tt_metal/hw/inc/compile_time_args.h

```diff
@@ -6,4 +6,5 @@
 
 namespace ckernel {
 #define get_compile_time_arg_val(arg_idx) KERNEL_COMPILE_TIME_ARG_##arg_idx
+#define kernel_compile_time_args(arg_idx) KERNEL_COMPILE_TIME_ARGS[arg_idx]
```

Comment:
I would recommend adding an additional line here to static asert that argidx is less than array size with a nice message